django/core/management/commands/runserver.py

```diff
@@ -181,10 +187,21 @@ def on_bind(self, server_port):
 
         now = datetime.now().strftime("%B %d, %Y - %X")
         version = self.get_version()
+
         print(
             f"{now}\n"
             f"Django version {version}, using settings {settings.SETTINGS_MODULE!r}\n"
             f"Starting development server at {self.protocol}://{addr}:{server_port}/\n"
             f"Quit the server with {quit_command}.",
             file=self.stdout,
         )
+        if not self.hide_production_warning_message:
+            self.stdout.write(
+                self.style.WARNING(
+                    "WARNING: This is a development server."
+                    " Do not use it in a production deployment. Use a production WSGI "
+                    "server instead.\n"
+                    "For more information on production servers see the documentation: "
+                    "https://docs.djangoproject.com/en/5.0/howto/deployment/"
```

Comment:
Maybe you could use `importlib.metadata` to determine the installed version? Django only supports 3.10 or better, `importlib.metadata` is no longer provisional now.

https://docs.python.org/3/library/importlib.metadata.html#module-importlib.metadata

Reply:
Why the installed version of python? I agree that I need to remove the hardcoded version of Django from the url, but I'm not sure why python.

Reply:
Not Python, Django.

```
./manage.py shell
>>> from importlib.metadata import version
>>> version("django")
'5.0.7'
>>> 
```

Reply:
Ah right. I will use the already present `version` variable which probably uses importlib under the hood.

Reply:
If you use "stable", the docs will redirect for you, save the additional complexity. It also means linking to the latest version of the deployment docs, which is probably better.

Reply:
Oh, that's much better. Thanks @RealOrangeOne , I'm going to use this in my own projects as well.

Reply:
Why it's better? It may not match to installed Django, e.g. some deployment options may not work, etc.

#18467